src/components/Focus/Focus.tsx

```diff
@@ -13,8 +13,22 @@ export const Focus = memo(function Focus({
   root,
 }: FocusProps) {
   useEffect(() => {
-    if (disabled || !root || root.querySelector('[autofocus]')) return;
-    focusFirstFocusableNode(root, false);
+    if (disabled || !root) {
+      return;
+    }
+
+    let node: HTMLElement | null;
+    if ('current' in root) {
+      node = (root as React.RefObject<HTMLElement>).current;
+    } else {
+      node = root as HTMLElement;
+    }
```

Comment:
Not a fan of this casting. Can you use a custom type guard here instead? https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Just pseudo coding here but something like this:

```js
function isRef(ref: React.RefObject<HTMLElement> | null): ref is React.RefObject<HTMLElement> {
  return (ref as React.RefObject<HTMLElement>).current !== undefined;
}
```

Reply:
Much cleaner thank you - made the change to use a type guard instead 👍